src/main/java/useCases/Calculator/Calculator.java

```diff
@@ -0,0 +1,107 @@
+package useCases.Calculator;
+
+import entities.Event;
+import entities.Task;
+import entities.User;
+import useCases.Calculator.helper.TimeConverter;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+abstract class Calculator {
+    private final int YEAR = 0;
+    private final int MONTH = 1;
+    private final int DAY = 2;
+    public Calculator(){}
+    /**
+     * Convert the string representing the unit of time passed in from the controller. For example, "Month" will
+     * bre converted to MONTH, which is 1.
+     * @param unit the unit of time (Day/Week/Month/Year)
+     * Precondition: unit must be "Month"/"Week"/"Day". Other strings are not considered as valid units.
+     * This private method is created to ensure that all calculators read inputs in the exact same way.
+     */
+     protected int unit_reader(String unit) {
+         if (Objects.equals(unit, "Month")) {
+             return MONTH;
+         } else if (Objects.equals(unit, "day")) {
+             return DAY;
+         }
+         else{
+             return 99;
```

Comment:
I don't understand the use of 99 here (magic number)! This could be refactored later down the line.